### PR TITLE
Impossible to import already exist keystore files

### PR DESCRIPTION
**Pull request by r8d8**
Process external keystore files (UTC / JSON) and store them locally.

### CODE REVIEW

**Comment by dulanov** [comment]
Please write more detailed comments, not just 'Added' or 'Fixed'.

**Comment by dulanov** [comment]
@r8d8 Do you want to return to this PR or will just close it?

File: src/keystore/serialize/crypto.rs [hunk 1]
```diff
@@ -156,7 +156,7 @@ impl Encodable for Crypto {
 }
 
 #[inline]
-fn decode_str<T: FromStr, D: Decoder>(d: &mut D) -> Result<T, D::Error>
+pub fn decode_str<T: FromStr, D: Decoder>(d: &mut D) -> Result<T, D::Error>
```
Comment (dulanov) [inline src/keystore/serialize/crypto.rs]: Why?

File: src/rpc/mod.rs [hunk 2]
```diff
@@ -179,6 +183,35 @@ pub fn start(addr: &SocketAddr, client_addr: &SocketAddr, base_path: Option<Path
...
+            Ok(ref v) if v.as_str().is_some() => {
+                let str = v.as_str().unwrap();
+                let kf = json::decode::<KeyFile>(str);
+
+                if kf.is_err() {
+                    return futures::done(Err(JsonRpcError::invalid_params("Invalid Keyfile \
+                                                                           data format")))
+                                   .boxed();
+                }
+                let res = kf.unwrap().flush("/tmp/", None);
```
Comment (dulanov) [inline src/rpc/mod.rs]: Not final solution?

File: src/rpc/mod.rs [hunk 3]
```diff
@@ -179,6 +183,35 @@ pub fn start(addr: &SocketAddr, client_addr: &SocketAddr, base_path: Option<Path
...
+
+                if res.is_err() {
+                    return futures::done(Err(JsonRpcError::invalid_params("Can't write Keyfile \
```
Comment (dulanov) [inline src/rpc/mod.rs]: Not invalide request parameters, just internal error, another one.

File: src/rpc/mod.rs (hunk 2 above)
Comment (r8d8) [inline src/rpc/mod.rs]: Yeap, not a final.

File: src/rpc/mod.rs [hunk 4]
```diff
@@ -179,39 +182,70 @@ pub fn start(addr: &SocketAddr, client_addr: &SocketAddr, base_path: Option<Path
...
+                let kf_res = json::decode::<KeyFile>(str);
 
-                if kf.is_err() {
+                if kf_res.is_err() {
                     return futures::done(Err(JsonRpcError::invalid_params("Invalid Keyfile \
                                                                            data format")))
                                    .boxed();
                 }
-                let res = kf.unwrap().flush("/tmp/", None);
+                let kf = kf_res.unwrap();
```
Comment (dulanov) [inline src/rpc/mod.rs]: Here IMHO better use `match`

Comment (r8d8) [inline src/rpc/mod.rs]: Done

**Pull request merged**